Domains/3-DisciplineOther/Hydraulics/SewerHydraulicAnalysis.ecschema.xml

```diff
@@ -168,6 +204,142 @@
+    <Source multiplicity="(1..1)" roleLabel="owns" polymorphic="false">
+      <Class class="PondRiser"/>
+    </Source>
+    <Target multiplicity="(1..1)" roleLabel="owned by" polymorphic="true">
+      <Class class="RiserTypeAspect"/>
+    </Target>
+  </ECRelationshipClass>
+ 
+  <ECEntityClass typeName="RiserPipeAspect" modifier="Sealed" displayLabel="Riser Pipe">
+    <BaseClass >RiserTypeAspect</BaseClass>
+    <ECProperty propertyName="OpeningDiameter" typeName="double" displayLabel="Diameter" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="The diameter of the opening at the top of the riser structure."/>
+  </ECEntityClass>
+ 
+  <ECEntityClass typeName="InletBoxAspect" modifier="Sealed" displayLabel="Inlet Box">
+    <BaseClass >RiserTypeAspect</BaseClass>
+    <ECProperty propertyName="WeirLength" typeName="double" displayLabel="Weir Length" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="Specify the active weir length for the opening at the top of the box."/>
+    <ECProperty propertyName="OrificeArea" typeName="double" displayLabel="Orifice Area" category="HydraulicData" kindOfQuantity="rru:AREA" description="Specify the active orifice area for the top opening when inlet box is submerged"/>
+  </ECEntityClass>
+
+	<ECEntityClass typeName="Orifice" modifier="Sealed" displayLabel="Orifice">
+    <BaseClass>FlowControlStructure</BaseClass>
+    <ECProperty propertyName="OrificeCoefficient" typeName="double" displayLabel="Orifice Coefficient" category="HydraulicData" description="Coefficient of flow for the orifice."/>
+    <ECProperty propertyName="OrificeOrientation" typeName="OrificeOrientation" displayLabel="Orientation" category="HydraulicData" description="Specify if the orifice oriented perpendicular or parallel to flow."/>
+    <ECProperty propertyName="OrificeShape" typeName="OrificeShape" displayLabel="Orifice Shape" category="HydraulicData" description="Specify the shape of the orifice"/>
+  </ECEntityClass>
+
+  <ECEntityClass typeName="OrificeShapeAspect" modifier="Abstract" displayLabel="OrificeShapeAspect">
+    <BaseClass >bis:ElementUniqueAspect</BaseClass>
+  </ECEntityClass>
+  
+	<ECRelationshipClass typeName="OrificeOwnsOrificeShapeAspect" strength="embedding" modifier="Sealed">
+    <BaseClass>bis:ElementOwnsUniqueAspect</BaseClass>
+    <Source multiplicity="(1..1)" roleLabel="owns" polymorphic="false">
+      <Class class="Orifice"/>
+    </Source>
+    <Target multiplicity="(1..1)" roleLabel="owned by" polymorphic="true">
+      <Class class="OrificeShapeAspect"/>
+    </Target>
+  </ECRelationshipClass>
+  
+	<ECEntityClass typeName="CircularOrificeAspect" modifier="Sealed" displayLabel="Circular Orifice">
+    <BaseClass>OrificeShapeAspect</BaseClass>
+    <ECProperty propertyName="Diameter" typeName="double" displayLabel="Diameter" category="HydraulicData" kindOfQuantity="rru:LENGTH" description="The diameter of the orifice opening"/>
```

Comment:
It seems to me that the "LENGTH" KoQ may not be the correct one for "Diameter" properties. Such KoQ is meant to use larger units by default (e.g. Meters or Feet). There is another KoQ in RoadRailUnits called "LENGTH_SHORT" which seems more appropriate.

Having said that, I can tell nobody has used the latter KoQ since its default presentation units are currently wrong (Meters and Feet-Inches). They should probably be mm and inches.

Reply:
@BenTWhite will address this detail in the next PR.